src/Routes/NavRoutes.js

```diff
@@ -0,0 +1,33 @@
+import { Routes, Route, Navigate } from "react-router-dom";
+import { Login, Signup } from "../features/auth";
+import { Home, Explore, Bookmark, SinglePage } from "../Pages";
+import { useSelector } from "react-redux";
+import { PrivateRoutes } from "./PrivateRoutes";
+
+const NavRoutes = () => {
+  const token = useSelector((state) => state.auth.token);
+  return (
+    <Routes>
+      <Route element={<PrivateRoutes />}>
+        <Route path="/" element={<Home />} />
+        <Route path="/explore" element={<Explore />} />
+        <Route path="/bookmark" element={<Bookmark />} />
+        <Route path="/single" element={<SinglePage />} />
+      </Route>
+
+      {!token ? (
+        <>
+          <Route path="/login" element={<Login />} />
+          <Route path="/signup" element={<Signup />} />
+        </>
+      ) : (
+        <>
+          <Route path="/login" element={<Navigate to="/" replace />} />
+          <Route path="/signup" element={<Navigate to="/" replace />} />
+        </>
+      )}
```

Comment:
This seems like extra code. Your PrivateRoutes does the same. You could just keep lines 20 and 21 and remove the rest.

Reply:
No this is deliberate because let's say if you are logged in still someone tries to reach the login or signup page by typing on the searchbar so in that case also the iser must go back to the home page.

Reply:
I am by manually typing /login or /signup

Reply:
I mean**